src/main/java/io/supertokens/session/Session.java

```diff
@@ -203,24 +214,33 @@ public static SessionInformationHolder getSession(Main main, @Nonnull String tok
                                 newAccessToken = AccessToken.createNewAccessTokenV1(main, accessToken.sessionHandle,
                                         accessToken.userId, accessToken.refreshTokenHash1, null,
                                         sessionInfo.userDataInJWT, accessToken.antiCsrfToken);
+                            } else if (AccessToken.getAccessTokenVersion(accessToken) == AccessToken.VERSION.V2) {
+                                assert accessToken.lmrt != null;
```

Comment:
The purpose of `lmrt` was to keep track of the "last manual regeneration time" -> whenever the session as manually regenerated (in case of re-authentication). But this is now not needed cause it will be a part of the session grant.

Therefore, we can get rid of this in access token version v3.

Reply:
But this can be another PR that is done on top of this one (after this is finalised)

Reply:
I think it should be in a separate PR.